google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -137,9 +138,10 @@ public CancellableRunnable createDeployRunner(
         wrapCallbackForUsageTracking(
             callback, deploymentConfiguration, source.getFile(), targetEnvironment));
 
-    if (targetEnvironment == AppEngineEnvironment.APP_ENGINE_STANDARD) {
+    if (targetEnvironment.isStandard()
+        || (targetEnvironment.isFlexible() && AppEngineUtil.isFlexCompat(project, source))) {
```

Comment:
it seems like environment is actually not the thing that decides which runner to use. I wonder if we couldn't change teh logic to be like

if (isStandardFrameworkApp(project, source) return newStandardFrameworkAppDeployer(..); else return create newNonStandardFrameworkAppDeployer(..);